novatel_gps_driver/src/nodelets/novatel_gps_nodelet.cpp

```diff
@@ -389,28 +391,32 @@ namespace novatel_gps_driver
 
           time_difference=ros::Time::now()-last_update_time_;
 
-          if (last_update_time_.isZero() || ((time_difference>ros::Duration(2.0)) && (status_gps==cav_msgs::DriverStatus::OPERATIONAL)))
+          if (last_update_time_.isZero() || ((time_difference>ros::Duration(1.0)) && (status_gps_==cav_msgs::DriverStatus::OPERATIONAL)))
           {
-            status_gps=cav_msgs::DriverStatus::OFF;
+            status_gps_=cav_msgs::DriverStatus::OFF;
           }
 
-          if (status_gps==cav_msgs::DriverStatus::OFF)
+          if (status_gps_==cav_msgs::DriverStatus::OFF)
           {
-           status_.status=cav_msgs::DriverStatus::OFF;
+           status_.status=status_gps_;
+           ROS_INFO("DriverStatus->OFF");
           }
-          else if (status_gps==cav_msgs::DriverStatus::OPERATIONAL)
+          else if (status_gps_==cav_msgs::DriverStatus::OPERATIONAL)
           {
-          status_.status=cav_msgs::DriverStatus::OPERATIONAL;
+          status_.status=status_gps_;
+          ROS_INFO("DriverStatus->OPERATIONAL");
           }
-          else if (status_gps==cav_msgs::DriverStatus::FAULT)
+          else if (status_gps_==cav_msgs::DriverStatus::FAULT)
           {
-          status_.status=cav_msgs::DriverStatus::FAULT;
+          status_.status=status_gps_;
+          ROS_INFO("DriverStatus->FAULT");
           }
-          else if(status_gps==cav_msgs::DriverStatus::DEGRADED)
+          else if(status_gps_==cav_msgs::DriverStatus::DEGRADED)
           {
-           status_.status=cav_msgs::DriverStatus::DEGRADED;
+           status_.status=status_gps_;
+           ROS_INFO("DriverStatus->DEGRADED");
```

Comment:
It would be better to only log when the state changes. Try comparing against previous state.